### ISSUE TITLE
Supercede `run_analysis.rb` with `buildstock_local`

### ISSUE THREAD

**Issue by joseph-robertson** [NREL/buildstockbatch#372]
**Is your feature request related to a problem? Please describe.**
See #349 for some context.

**Describe the solution you'd like**
The `buildstock_local` utility to have all the features of `run_analysis.rb` that are nice to have. For example, the ability to run only sets of specified building IDs.

It also appears that the `buildstock_local` utility requires that you update your path env var to point to the correct version of openstudio. Idea from @shorowit: Since the OS CLI now supports python, maybe bsb could be updated so that if you use `openstudio buildstock_local ...` instead of `python buildstock_local`, it's smart enough to use that openstudio.

**Goals**
Adopt `buildstock_local` in place of `run_analysis.rb` - it's "official" and means a whole lot less code to maintain in the resstock repo.

**Comment by shorowit** [comment]
> It also appears that the `buildstock_local` utility requires that you update your path env var to point to the correct version of openstudio. Idea from @shorowit: Since the OS CLI now supports python, maybe bsb could be updated so that if you use `openstudio buildstock_local ...` instead of `python buildstock_local`, it's smart enough to use that openstudio.

I assume it's the python equivalent of something like this?
```
begin
  require 'openstudio'
  cli_path = OpenStudio.getOpenStudioCLI
rescue
  cli_path = ENV['OPENSTUDIO_EXE']
end
```

**Comment by afontani** [comment]
From @joseph-robertson : Some potential features or benefits to run_analysis.rb
- [x] Only needs OS CLI (can we reduce the overhead?)
- [ ] Only run a single or set of building IDs (more specific building model debugging)
- [ ] Collect the warnings into a single file

**Comment by afontani** [comment]
One more item in run_analysis.rb:

**Comment by joseph-robertson** [comment]
@nmerket On the first (checked) item above, I'm confused. Something was changed/updated in buildstockbatch such that you can now use `openstudio buildstock_local`?

**Comment by nmerket** [comment]
No, `buildstock_local` is still a python app. But it calls `openstudio` directly to run each simulation. We don't have any plans (or a way forward) to run `buildstock_local` from openstudio.

**Comment by shorowit** [comment]
@nmerket `openstudio` can now run python scripts (just like it can run ruby scripts).

**Comment by nmerket** [comment]
Yes, but there are [a lot of dependencies](https://github.com/NREL/buildstockbatch/blob/develop/setup.py#L32-L47) in buildstockbatch. Several have compiled components. If it's anything like how ruby is shipped with openstudio, I'm dubious that we'll ever be able to run buildstockbatch _from_ openstudio.

**Comment by shorowit** [comment]
Ah, right. So that means running `buildstock_local` will always require more setup than `run_analysis.rb`.

**Comment by rajeee** [comment]
What's the value of being able to run buildstockbatch _from_ openstudio? Not having to setup buildstockbatch separately?

**Comment by joseph-robertson** [comment]
Yes. From my perspective, with `run_analysis.rb` I don't need to: `source activate xxx` anything, `pip install xxx` anything, or change any paths.

**Comment by shorowit** [comment]
To draw a parallel, the user can run `run_analysis.rb` with either ruby or openstudio. But by using openstudio, the user doesn't have to:
1. Separately install ruby and any gems
2. Tell the script where openstudio is installed

Plus there are fewer things that can go wrong -- wrong version of ruby, trouble compiling gems, etc. (And they already have openstudio since it's needed to run the simulations.)

**Comment by nmerket** [comment]
How did getting nokogiri installed with openstudio go? Now imagine that with numpy, pandas, dask, pyarrow, lxml, s3fs, boto3, and more. Unless openstudio comes with a fully functional pip package manager (it might for all I know), we'd either need to preload that full stack of libraries in openstudio or rewrite buildstockbatch from scratch with only modules in the standard library. I'm not sure that's feasible or even possible. There's several things that buildstockbatch does that run_analysis.rb doesn't do, most notably timeseries aggregation, that depend heavily on those libraries.

**Comment by rajeee** [comment]
I don't quite understand how openstudio handles package dependencies. For the case of ruby, @shorowit  you said using openstudio directly means the user doesn't have to install any gems. Is the assumption there that the script is only using the gems that comes preinstalled with openstudio? 

And I presume same is the case with python where python and some set of packages are shipped with openstudio?

**Comment by joseph-robertson** [comment]
Probably getting into @kbenne's territory here.

**Comment by shorowit** [comment]
Yes, openstudio comes preinstalled with various gems (run `openstudio gem_list`). We generally write ruby scripts to rely only on those gems, as that keeps it simple for the end user. That said, I think you can point openstudio at additional gems to use.

Support for python in openstudio is pretty new. I think it only comes with the standard library today; I don't know what the longer term plans are.
